torch/fx/experimental/sym_node.py

```diff
@@ -1134,6 +1202,13 @@ def binary_magic_impl(self, other):
                     out = Mod(self.expr, other.expr)
                 else:
                     out = PythonMod(self.expr, other.expr)
+            elif method == "add":
+                (optimized_summation, out) = func(
+                    self.expr,
+                    other.expr,
+                    getattr(self, "_optimized_summation", False),
+                    getattr(other, "_optimized_summation", False),
```

Comment:
We discussed this in person, where you defended the ad hoc getattr/setattr because it was only set and accessed from two places. I think my preferred way of documenting this sort of situation is to have a `# Note [blah blah blah]` in one location describing the situation (probably the comment below), and then referencing that note consistently at both use sites.

The important thing to document, which is not directly documented at either of these sites, is what exactly the invariant specified by optimized summation is.

Reply:
Yeah, I think I would even be happy with "this is a subclass of Add and is identical to Add in all respects except it respects the optimized summation invariant". This would probably have some annoying side effects in other parts of the code but from a layering perspective it's much cleaner.

Reply:
i tried the subclass it did not work, because we could get an add to co-live with custom add and they dont compare to be equal